prow/cmd/gangway/main.go

```diff
@@ -156,26 +157,35 @@ func main() {
 		}
 	}
 
-	gitClient, err := o.github.GitClientFactory(o.cookiefilePath, &o.config.InRepoConfigCacheDirBase, o.dryRun, false)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error getting Git client.")
+	gw := gangway.Gangway{
+		ConfigAgent:   configAgent,
+		ProwJobClient: prowjobClient,
 	}
 
-	cacheGetter, err := config.NewInRepoConfigCache(o.config.InRepoConfigCacheSize, configAgent, gitClient)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error creating InRepoConfigCacheGetter.")
+	// InRepoConfig getter.
+	if o.config.MoonrakerAddress != "" {
+		moonrakerClient, err := moonraker.NewClient(o.config.MoonrakerAddress, 10*time.Second)
+		if err != nil {
+			logrus.WithError(err).Fatal("Error getting Moonraker client.")
+		}
+		gw.InRepoConfigGetter = moonrakerClient
+	} else {
+		gitClient, err := o.github.GitClientFactory(o.cookiefilePath, &o.config.InRepoConfigCacheDirBase, o.dryRun, false)
+		if err != nil {
+			logrus.WithError(err).Fatal("Error getting Git client.")
+		}
+		ircc, err := config.NewInRepoConfigCache(o.config.InRepoConfigCacheSize, configAgent, gitClient)
+		if err != nil {
+			logrus.WithError(err).Fatal("Error creating InRepoConfigCache.")
+		}
+		gw.InRepoConfigGetter = ircc
 	}
```

Comment:
You probably felt it too, but there's a considerable amount of repetition across the services. I wonder if it would make sense to put this logic in inrepoconfig.go?

Certainly fine as-is though.

Reply:
Might be worth revisiting as a follow-up item. Maybe I can create a function that takes in a pointer-to-an-interface as an argument and assigns the interface inside it?

Reply:
This cleanup should be done with a larger cleanup of the necessarily related TODO item here: https://github.com/kubernetes/test-infra/blob/293bdb865275609b1d5f8878026aaafb47409e8c/prow/flagutil/github.go#L339